### ISSUE TITLE
cannot create event unhalted_core_cycles

### ISSUE THREAD

**Issue by ronghongbo** [google/gematria#117]
Hello, I checked out the latest gematria (54097142a71fe8d6fcec8d2910949cb5ba1c60a3) for Ubuntu 22.04.4 LTS in WSL on a 12th Gen Intel(R) Core(TM) i7-1270P CPU. 

I made no change except: 
```
requirements.in
-tensorflow-probability>=0.19.0
+tensorflow-probability>=0.23.0
-tensorflow>=2.11.0; sys_platform=='linux'
+tensorflow>=2.15.1; sys_platform=='linux'
```
Then build, and test:
```
(env) (base) ~/gematria$   env USE_BAZEL_VERSION=6.4.0 ../bazelisk-linux-amd64 test ...
.....
FAIL: //gematria/datasets:find_accessed_addrs_exegesis_test (see /home/hrong1/.cache/bazel/_bazel_hrong1/32246067180bfaeac7e17e4449bcdc84/execroot/com_google_gematria/bazel-out/k8-fastbuild/testlogs/gematria/datasets/find_accessed_addrs_exegesis_test/test.log)
[6,387 / 6,636] 22 / 52 tests, 1 failed; 16 actions running; last test: //gematria/llvm/python:canonicalizer_test
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/compiler/plugin_pb2.py:
external/com_google_protobuf/.: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/any_pb2.py:
external/com_google_protobuf/.: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/duration_pb2.py:
external/com_google_protobuf/.: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/descriptor_pb2.py:
external/com_google_protobuf/.: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/api_pb2.py:
external/com_google_protobuf/.: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/empty_pb2.py:
external/com_google_protobuf/.: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/field_mask_pb2.py:
external/com_google_protobuf/.: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/wrappers_pb2.py:
external/com_google_protobuf/.: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/source_context_pb2.py:
external/com_google_protobuf/.: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/struct_pb2.py:
external/com_google_protobuf/.: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/timestamp_pb2.py:
external/com_google_protobuf/.: warning: directory does not exist.
INFO: From ProtoCompile external/com_google_protobuf/python/google/protobuf/type_pb2.py:
external/com_google_protobuf/.: warning: directory does not exist.
```
The log file of the failed test: `/home/hrong1/.cache/bazel/_bazel_hrong1/32246067180bfaeac7e17e4449bcdc84/execroot/com_google_gematria/bazel-out/k8-fastbuild/testlogs/gematria/datasets/find_accessed_addrs_exegesis_test/test.log`:
```
Executing tests from //gematria/datasets:find_accessed_addrs_exegesis_test
-----------------------------------------------------------------------------
Running main() from gmock_main.cc
[==========] Running 5 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 5 tests from FindAccessedAddrsExegesisTest
[ RUN      ] FindAccessedAddrsExegesisTest.ExegesisNoAccess
event not found - cannot create event unhalted_core_cycles
gematria/datasets/find_accessed_addrs_exegesis_test.cc:91: Failure
Value of: static_cast<bool>(AddrsOrErr)
  Actual: false
Expected: true
```
Any idea? Thanks!
Hongbo

**Comment by boomanaiden154** [comment]
Not too familiar with WSL, but doesn't it utilize a virtual machine to run Linux? Even if it doesn't there might be some oddities with the perf subsystem. I would expect you need bare metal to get this to pass (although the test itself probably shouldn't run on machines without performance counters).

**Comment by ronghongbo** [comment]
Thanks for the suggestion, Aiden. Yes, a virtual machine could be a problem. Will try a native build.

**Comment by ondrasej** [comment]
+1 to what Aiden said, this version of the code seems to depend on access to PMU (hardware perf counters) which are typically disabled or severely limited in virtual machines. Please try running this on a physical machine. If you're planning to build a data set, I'd recommend using a physical Linux machine for the measurements too, to avoid any limitations or additional noise coming from the use of containers/VMs.

Also relevant: we also have another version of this accessed addresses code that doesn't need perf counters to run in `//gematria/datasets:find_accessed_addrs` (that will get even more useful once #119 is merged).

**Comment by ronghongbo** [comment]
Totally agree. Thanks for the suggestion of both of you. I'm closing the issue as it is specific to a virtual machine. I tried Gematria on a native Linux machine, but got some errors with installing pthread pool library when compiling tflite, and stopped there temporarily. Will return to that native Linux machine when it is time to do some big scale experiments.

**Comment by ronghongbo** [comment]
Confirmed: on a native Linux machine, the test works fine. So the issue was due to the WSL virtual machine.

**Comment by boomanaiden154** [comment]
Makes sense.

By the way, the `pthreadpool` dependency for the TFLite should be going away soonish.

**Comment by ronghongbo** [comment]
Getting rid of pthreadpool would be great. It seems not a new issue, but has persisted for quite a long time: https://github.com/tensorflow/tensorflow/issues/57658
